### ISSUE TITLE
Docerina fails on package structures that has non-package folders

### ISSUE THREAD

**Issue by rasika** [ballerina-platform/ballerina-lang#10568]
**Description:**
Consider a project that has following folder structure;
```
.
├── Ballerina.toml
├── foo
│   ├── bar.bal
│   └── resources
│       ├── sample.bal
│       └── sample.png
├── main.bal
├── model
│   ├── card.bal
│   └── trick.bal
└── resources [docs generation fails here]
    └── sample.png
```
When you try to generate docs for the whole ballerina project with the command; `ballerina doc`. Docerina should fail with following message;

```
docerina: API documentation generation for sources - [resources, foo, model]
	could not find package abc/resources:0.0.1
```

**Observations:**
This reason to stop doc generation in the middle is; Compiler is throwing a `BLangCompilerException` in [here](https://github.com/ballerina-platform/ballerina-lang/blob/18f94b36b04cdafddc16979665e3d6ee4a44c267/misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java#L299). However; catching `BLangCompilerException` in [the catch block](https://github.com/ballerina-platform/ballerina-lang/blob/18f94b36b04cdafddc16979665e3d6ee4a44c267/misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java#L301) and continuing causes another issue; that `ballerina doc` command does not exit after doc generation(due to unterminated non-daemon threads initiated by the Compiling process). 

Adding a `System.exit(0)` [after doc generation](https://github.com/ballerina-platform/ballerina-lang/blob/18f94b36b04cdafddc16979665e3d6ee4a44c267/misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java#L283) won't help since it kills maven-doc-generation plugin while building the ballerina repo.

Thread dump after java main method exit;

```
2018-09-27 08:04:14
Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.181-b13 mixed mode):

"Attach Listener" #20 daemon prio=9 os_prio=31 tid=0x00007fe91f013800 nid=0x5907 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"DestroyJavaVM" #19 prio=5 os_prio=31 tid=0x00007fe91d622000 nid=0x1a03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"worker-thread-pool-1" #17 prio=5 os_prio=31 tid=0x00007fe916954800 nid=0xa203 waiting on condition [0x000070000d353000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x0000000780e17488> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"nioEventLoopGroup-7-1" #16 prio=10 os_prio=31 tid=0x00007fe91bb67000 nid=0xa403 runnable [0x000070000d250000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
	at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
	at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x00000007806777c8> (a io.netty.channel.nio.SelectedSelectionKeySet)
	- locked <0x00000007806788d0> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000007806787f0> (a sun.nio.ch.KQueueSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at io.netty.channel.nio.SelectedSelectionKeySetSelector.select(SelectedSelectionKeySetSelector.java:62)
	at io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:753)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:409)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"Timer-0" #15 daemon prio=5 os_prio=31 tid=0x00007fe9195be000 nid=0x1007 in Object.wait() [0x000070000d14d000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000007805ba300> (a java.util.TaskQueue)
	at java.util.TimerThread.mainLoop(Timer.java:552)
	- locked <0x00000007805ba300> (a java.util.TaskQueue)
	at java.util.TimerThread.run(Timer.java:505)

   Locked ownable synchronizers:
	- None

"pool-1-thread-1" #14 prio=5 os_prio=31 tid=0x00007fe91cff7000 nid=0xf07 waiting on condition [0x000070000ba05000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000078065bfc8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"Service Thread" #12 daemon prio=9 os_prio=31 tid=0x00007fe917804000 nid=0x5603 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C1 CompilerThread3" #11 daemon prio=9 os_prio=31 tid=0x00007fe915849000 nid=0xa903 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread2" #10 daemon prio=9 os_prio=31 tid=0x00007fe91583a800 nid=0x4403 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread1" #9 daemon prio=9 os_prio=31 tid=0x00007fe915848000 nid=0x4303 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread0" #8 daemon prio=9 os_prio=31 tid=0x00007fe917800800 nid=0x4703 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"JDWP Command Reader" #7 daemon prio=10 os_prio=31 tid=0x00007fe918800000 nid=0x4203 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"JDWP Event Helper Thread" #6 daemon prio=10 os_prio=31 tid=0x00007fe91686a800 nid=0x4103 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"JDWP Transport Listener: dt_socket" #5 daemon prio=10 os_prio=31 tid=0x00007fe916844800 nid=0x4a07 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Signal Dispatcher" #4 daemon prio=9 os_prio=31 tid=0x00007fe915838800 nid=0x3e0b runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Finalizer" #3 daemon prio=8 os_prio=31 tid=0x00007fe916011000 nid=0x3403 in Object.wait() [0x000070000c62c000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000078065caf0> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
	- locked <0x000000078065caf0> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:216)

   Locked ownable synchronizers:
	- None

"Reference Handler" #2 daemon prio=10 os_prio=31 tid=0x00007fe916010800 nid=0x4e03 in Object.wait() [0x000070000c529000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000078065cb30> (a java.lang.ref.Reference$Lock)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x000000078065cb30> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

   Locked ownable synchronizers:
	- None

"VM Thread" os_prio=31 tid=0x00007fe916009800 nid=0x3203 runnable

"GC task thread#0 (ParallelGC)" os_prio=31 tid=0x00007fe916004800 nid=0x1e07 runnable

"GC task thread#1 (ParallelGC)" os_prio=31 tid=0x00007fe916005000 nid=0x1f03 runnable

"GC task thread#2 (ParallelGC)" os_prio=31 tid=0x00007fe916005800 nid=0x5403 runnable

"GC task thread#3 (ParallelGC)" os_prio=31 tid=0x00007fe916006000 nid=0x2c03 runnable

"GC task thread#4 (ParallelGC)" os_prio=31 tid=0x00007fe916007000 nid=0x2e03 runnable

"GC task thread#5 (ParallelGC)" os_prio=31 tid=0x00007fe916007800 nid=0x5303 runnable

"GC task thread#6 (ParallelGC)" os_prio=31 tid=0x00007fe916008000 nid=0x3003 runnable

"GC task thread#7 (ParallelGC)" os_prio=31 tid=0x00007fe916834000 nid=0x5003 runnable

"VM Periodic Task Thread" os_prio=31 tid=0x00007fe916014000 nid=0x5703 waiting on condition

JNI global references: 15899
```

Seems like netty threads should be daemonized as in https://github.com/netty/netty/issues/2832

**Comment by rasika** [comment]
Closing since not reproducible in the latest.
